packages/mongoose/src/hooks.ts

```diff
@@ -36,20 +36,22 @@ type Hook =
 	| 'findOneAndDelete'
 	| 'findOneAndRemove';
 
+type ResultType = Document | Document[];
```

Comment:
can you change this to be 'QueryResultType'?

'ResultType' is a bit too generic

Reply:
Yep I can